### PR TITLE
UIOR-1018 Provide local translations to `ControlledVocab`

### PR DESCRIPTION
**Pull request by usavkov-epam**
## Purpose

https://issues.folio.org/browse/UIOR-1018
## Pre-Merge Checklist
Before merging this PR, please go through the following list and take appropriate actions.
- [x] I've added appropriate record to the CHANGELOG.md
- Does this PR meet or exceed the expected quality standards?
  - [ ] Code coverage on new code is 80% or greater
  - [x] Duplications on new code is 3% or less
  - [x] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] If any API-related changes - okapi interfaces and permissions are reviewed/changed correspondingly
  - [x] There are no breaking changes in this PR.

If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

**Comment by Terala-Priyanka** [comment]
Could you update changelog read me file as well?

**Review by Terala-Priyanka — CHANGES_REQUESTED** [review]
Could you update changelog read me file as well?

**Comment by usavkov-epam** [comment]
> Could you update changelog read me file as well?

Oh, good catch, thank you!

**Review by Terala-Priyanka — APPROVED** [review]

File: translations/ui-orders/en.json [hunk 1]
```diff
@@ -726,8 +741,12 @@
   "settings.setPOLInesLimit": "Set purchase order lines limit",
 
   "settings.acquisitionMethods": "Acquisition methods",
-  "settings.acquisitionMethods.singular": "Acquisition method",
```
Comment (zburke) [inline translations/ui-orders/en.json:729]: You can't remove this. It's still in use via [the `objectLabel` prop](https://github.com/folio-org/ui-orders/blob/f0194a92d981635e8eec9496d99fd299d0a67a94/src/settings/AcquisitionMethods/AcquisitionMethods.js#L55)

**Review by zburke — CHANGES_REQUESTED** [review]
Looks good, except the translation key `settings.acquisitionMethods.singular` needs to be preserved.

**Review by zburke — COMMENTED** [review]
Additionally, I think you'll need to bump the minimum version of `stripes-acq-compents` in your `package.json` in order to make sure `getControlledVocabTranslations` is part of it. The most recent release was `3.3.0` so presumably the minimum version here will be `3.4.0`.

File: translations/ui-orders/en.json (hunk 1 above)
Comment (usavkov-epam) [reply translations/ui-orders/en.json:729]: You are absolutely right, thanks! Updated

**Comment by usavkov-epam** [comment]
> Additionally, I think you'll need to bump the minimum version of `stripes-acq-compents` in your `package.json` in order to make sure `getControlledVocabTranslations` is part of it. The most recent release was `3.3.0` so presumably the minimum version here will be `3.4.0`.

Thanks, @zburke. We usually update this version, when preparing release for modules (after `stripes-acq-components` release)

**Comment by zburke** [comment]
> We usually update this version, when preparing release for modules (after stripes-acq-components release)

That's wrong. 

`<rant>`
The minor version of every module (whether an app or a shared library) should be bumped immediately after release. If you were to commit this PR right now and then publish a release from tip of master, the platform build would succeed (because ui-orders claims to only need stripes-acq-components 3.3.0, which exists) but only because this app lies about its needs (in fact, ui-orders really needs stripes-acq-components 3.4.0, which does not exist). In other words, waiting until release-time to bump the version number has several bad consequences:

1. It squeezes the amount of time apps have to bump their dependencies, forcing them to wait until after their dependencies are formally released.
2. It allows changes to accumulate on the "wrong" version, e.g. `getControlledVocabTranslations` doesn't exist in v3.3.0 but it sure looks like it does if you look at the commit log since that's the version present in `package.json` when that feature was added.
3. It prevents us from detecting this bug at build time. For locales that use gendered articles (e.g. German with "das", "der", and "die"), this PR will be considered a bug fix because it allows correct grammar to be used. Thus, it's easy to imagine this PR being cherry-picked for a patch release. If we published a hotfix with no other changes than this one, tests would pass (because test builds run against the CI repository) and the platform would build succeed (because this `package.json` only claims to need v3.3.0) but this page would blow up when somebody actually tried to use it. Instead, if stripes-acq-components had correctly bumped its version to 3.4.0, and ui-orders had correctly bumped its dependency, the platform build would fail and we'd catch the error before it made it into production. 

`</rant>`

This may feel beyond the scope of this particular PR, and it may feel like I'm ranting about an undocumented versioning policy. I'm sorry if that's the case. I will think about how/where to document these conventions and would welcome feedback about the best way to help people learn about them. 

In the mean time, we need to get the version on `stripes-acq-components#master` bumped to `3.4.0` and likewise bump the minimum version in `package.json` here. 

CC: @NikitaSedyx

**Comment by NikitaSedyx** [comment]
@zburke probably you found a wide issue across mostly all modules (I believe except stripes-*), versions are bumped before the release to be matched with the release version

`Working towards the next release` part in release procedure doesn't say that package.json version should be bumped after release is done

do we really need to do this having different repositories prod and non-prod? it will complicate a bit after release flow (even react guys don't do it, version is incremented right before the release)

**Comment by NikitaSedyx** [comment]
> It squeezes the amount of time apps have to bump their dependencies, forcing them to wait until after their dependencies are formally released.

I would agree, but we have order of releases so it shouldn't be an issue, and based on previous release cycles it's not an issue

> It prevents us from detecting this bug at build time. For locales that use gendered articles (e.g. German with "das", "der", and "die"), this PR will be considered a bug fix because it allows correct grammar to be used. Thus, it's easy to imagine this PR being cherry-picked for a patch release. If we published a hotfix with no other changes than this one, tests would pass (because test builds run against the CI repository) and the platform would build succeed (because this package.json only claims to need v3.3.0) but this page would blow up when somebody actually tried to use it. Instead, if stripes-acq-components had correctly bumped its version to 3.4.0, and ui-orders had correctly bumped its dependency, the platform build would fail and we'd catch the error before it made it into production.

agree, and it's team responsibility to correctly cherry-pick and release properly, we connect tickets for this purpose


> It allows changes to accumulate on the "wrong" version, e.g. getControlledVocabTranslations doesn't exist in v3.3.0 but it sure looks like it does if you look at the commit log since that's the version present in package.json when that feature was added.

can't fully agree, as we use ~, it means that `getControlledVocabTranslations` can be in any patch versions and our folioci versions are patch + in stripes-acq-components it's clearly described that it will be part of next version

**Comment by zburke** [comment]
@NikitaSedyx, you make some good points, especially that our documentation is inadequate. My problem with this particular PR is that if you look at it in isolation there is no way to tell that cherry-picking it and publishing it in a patch release will cause a runtime error. 

If you think we should manage versions differently, then let's get it in front of more eyes: start a thread on Slack#stripes-architecture and then let's plan a discussion at an upcoming weekly meeting to confirm the plan, document the decision, and then get all teams following the same pattern.

**Comment by NikitaSedyx** [comment]
@zburke 
> My problem with this particular PR is that if you look at it in isolation there is no way to tell that cherry-picking it and publishing it in a patch release will cause a runtime error.

fully agree with this point, only responsible team can know about it

> If you think we should manage versions differently

I'd prefer to keep it as is (process developers follow currently and matched with documentation), but it doesn't solve point above, so we can start release procedure discussion to adjust it or in addition to jira tickets keep references in PRs so someone else (not only members from specific team) can understand that changes rely on new functionality from another repo

**Comment by zburke** [comment]
Sorry, @NikitaSedyx, I cannot approve this PR as-is. If we're going to ask for community review of PRs by @folio-org/fe-tl-reviewers, then we need community standards so that all parties have the same expectations. We can either meet and agree to a standard or you can remove @folio-org/fe-tl-reviewers as a reviewer.

**Comment by NikitaSedyx** [comment]
@zburke to not block batch of requests, @usavkov-epam will bump up version, and I believe we should meet with FE leads to discuss - it will take time as usual

**Comment by usavkov-epam** [comment]
@zburke `stripes-acq-components` version was bumped up to `3.4.0`, please take a look, thanks!

**Review by zburke — APPROVED** [review]

**Pull request merged**